Centaurus/Alpha/AlphaStartup.cs

```diff
@@ -0,0 +1,165 @@
+using System;
+using System.Net.WebSockets;
+using Centaurus.Domain;
+using Centaurus.Models;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using NLog;
+using System.Linq;
+using System.Threading.Tasks;
+using Centaurus.DAL.Mongo;
+using Microsoft.Extensions.Logging;
+using NLog.Web;
+using System.IO;
+
+namespace Centaurus
+{
+    public class AlphaStartup : IStartup<AlphaSettings>
+    {
+        static Logger logger = LogManager.GetCurrentClassLogger();
+        private IHost host;
+
+        public void Run(AlphaSettings settings)
+        {
+            ConfigureConstellation(settings);
+
+            host = CreateHostBuilder(settings).Build();
+            _ = host.RunAsync();
+        }
+
+        public void Shutdown()
+        {
+            ConnectionManager.CloseAllConnections();
+            host.StopAsync().Wait();
+        }
+
+        #region Private Members
+
+        private void ConfigureConstellation(AlphaSettings settings)
+        {
+            Global.Init(settings, new MongoStorage());
+
+            Global.AppState.StateChanged += Current_StateChanged;
+
+            MessageHandlers<AlphaWebSocketConnection>.Init();
+        }
+
+        private void Current_StateChanged(object sender, ApplicationState e)
+        {
+            if (e == ApplicationState.Failed)
+            {
+                logger.Error("Application failed");
```

Comment:
It would be nice to print at least something here - a reason or error description. Is it problematic to retrieve last error here? Or maybe we have already logged the error somewhere earlier?

Reply:
All exceptions that leads to application failed state are logged before this event called. But maybe we need to extend the event and add source/reason of state change. Need to discuss it.